scripts/west_commands/runners/mdb.py

```diff
@@ -0,0 +1,107 @@
+
+    @classmethod
+    def name(cls):
+        return 'mdb'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash', 'debug'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--jtag', default='digilent',
+                            help='choose the jtag interface')
+        parser.add_argument('--cores', default=1,
+                            help='choose the cores to debug')
+        parser.add_argument('--nsim_args', default='',
+                            help='if given, arguments for nsim simulator')
+        parser.add_argument('--dig-device', default='',
+                            help='''choose the digilent device when
+                             --jtag=digilent''')
+
+    @classmethod
+    def create(cls, cfg, args):
+        return MdbBinaryRunner(
+            cfg,
+            cores=args.cores,
+            jtag=args.jtag,
+            nsim_args=args.nsim_args,
+            dig_device=args.dig_device
+            )
+
+    def do_run(self, command, **kwargs):
+        self.require(self.commander)
+
+        mdb_basic_options = ['-nooptions', '-nogoifmain',
+                            '-toggle=include_local_symbols=1']
+
+        # nsim
+        if self.nsim_args != '':
+            mdb_target = ['-nsim', '@' + self.nsim_args]
+        # hardware target
+        else:
+            if self.jtag == 'digilent':
+                mdb_target = ['-digilent', self.dig_device]
+            else:
+                # \todo: add support of other debuggers
+                mdb_target = ['']
+
+        if command == 'flash':
+            mdb_run = ['-run', '-cmd=-nowaitq run', '-cmd=quit','-cl']
+        elif command == 'debug':
+            # use mdb gui to debug
+            mdb_run = ['-OKN']
+        else:
+            raise ValueError('unsupported command {}'.command)
+
+        if self.cores == 1:
+            # single core's md command is different with multicores
+            mdb_cmd = ([self.commander] + mdb_basic_options + mdb_target +
+                       mdb_run + [self.elf_name])
+        elif self.cores <= 4:
+            mdb_multifiles='-multifiles='
+            for i in range(self.cores):
+                mdb_sub_cmd = ([self.commander] +
+                            ['-pset={}'.format(i+1),
+                             '-psetname=core{}'.format(i),
+            # -prop=download=2 is used for SMP application debug, only the 1st
+            # core will download the shared image.
+                             ('-prop=download=2' if i > 0 else '')] +
+                             mdb_basic_options + mdb_target + [self.elf_name])
+                self.check_call(mdb_sub_cmd)
+                mdb_multifiles += (',core{}'.format(i) if i > 0 else 'core{}'.format(i))
+
+            mdb_cmd = ([self.commander] + [mdb_multifiles] + mdb_run)
+        else:
+            raise ValueError('unsupported cores {}'.format(self.cores))
+
+        #print(mdb_cmd)
```

Comment:
Please remove commented-out debug code. There's no need for it: running `west -v flash` will log all the arguments to `self.check_call` in the next line. You can also set `_DRY_RUN = True` in runners/core.py for debugging.

Reply:
Thanks, removed